netutils/netinit/netinit.c

```diff
@@ -357,6 +367,86 @@ static void netinit_set_macaddr(void)
 #  define netinit_set_macaddr()
 #endif
 
+#ifdef CONFIG_FSUTILS_IPCFG
+/****************************************************************************
+ * Name: netinit_get_ipaddrs
+ *
+ * Description:
+ *   Setup IP addresses.
+ *
+ *   For 6LoWPAN, the IP address derives from the MAC address.  Setting it
+ *   to any user provided value is asking for trouble.
+ *
+ ****************************************************************************/
+
+static int netinit_get_ipaddrs(void)
+{
+#ifdef CONFIG_NETINIT_DHCPC
+  bool use_dhcp   = false;
+#endif
+  bool use_static = false;
+#ifdef CONFIG_NET_IPv4
+  struct in_addr addr;
+  int ret;
+
+  ret = ipcfg_read(NET_DEVNAME, &g_netconf);
```

Comment:
why not call ipcfg_read for IPv6?

Reply:
I do not have a way to test it.  Can you test it?, it it works we can move the call up out of the #ifdef CONFIG_NET_IPv4

Reply:
There is no support for IPv6 in ipcfg_read/write at present.  That would have to be extended first.  There are some semi-standard IPv6 definitions (although many of them do not apply):

https://www.thegeekdiary.com/understanding-the-network-interface-configuration-file-etc-sysconfig-network-scripts-ifcfg-eth/
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/s1-networkscripts-interfaces

But it is not clear how to integrate IPv6 support into the binary interface without useless explosive growth since the binary form contains all possible settings. IPv6 support would require some additional design work. Certainly, if IPv6 support is added, that will delay this so that it misses the Window for the 10.0 release.  So, the options are IPv4 only for now (10.0) or with IPv6 support in 10.1.

Reply:
I totally agree. I will push the minor changes in a bit....

Reply:
> 
> 
> I totally agree. I will push the minor changes in a bit....

Don't get me wrong.  I think IPv6 support is critical, at least for static IP assignment (IPv6 does not use DHCP but has a separate auto-configuration feature).  The only issue if we merge this now and incorporate another PR later, or if we essentially close this and delay merging to add the IPv6 support.

Delaying the merge provides better assurance that the IPv6 support will not be swept under the rug but pushes the feature to 10.1.  Really your choice.  I have no strong opinion.  I am just trying to point features.

Reply:
I think this should come in as is and be in the 10.0 release. I will commit to help implement IPv6 in the 10.1 release.   I am not set up currently to test IPv6.